citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -211,7 +211,13 @@ public ICitation updateCitationFromZotero(IUser user, String groupId, String ite
             throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException {
         Optional<ICitationGroup> groupOptional = groupRepository.findFirstByGroupId(new Long(groupId));
         if (!groupOptional.isPresent()) {
-            throw new GroupDoesNotExistException("Group with id " + groupId + " does not exist.");
+            ICitationGroup group = groupManager.getGroup(user, groupId);
+            if (group == null) {
+                throw new GroupDoesNotExistException("Group with id " + groupId + " does not exist.");
+            } else {
+                asyncCitationProcessor.sync(user, groupId, group.getContentVersion(), null);
+                return null;
```

Comment:
never mind I was confused here; but I would not return null here. Just trigger the sync and retrieve the citation to be displayed. Otherwise, if the group is very big, the user has to wait potentially hours till they can see the citation.

Reply:
In that case, I'll remove the loading page since there is no need for it.